### ISSUE TITLE
Inf metric scores

### ISSUE THREAD

**Issue by nicolas-mng** [YefanZhou/TempBalance#2]
Hello,
I am experimenting with your library. I have realised that some layers end up having an infinite metric score. This has for effect to squeeze the learning rates towards the extremes, i.e. the learning rate multipliers are all 0.5 or 1.5. Have you encountered this scenario? How would you deal with these? I've tried to replace the infinite scores by the median to cancel their effects and have a better distribution of the LR factors in the range 0.5, 1.5 but in my use case, the accuracy ended up being worse than when using no TempBalance at all.
Many thanks!

**Comment by YefanZhou** [comment]
Hi, we didn't encounter this scenario in our [experiments](https://github.com/YefanZhou/TempBalance/tree/main#compare-tempbalance-tb-to-cal-and-snr). We found that if one layer only has 1 or 2 eigenvalues, the metric alpha score can be infinite. I added a variable `eigs_num` to help check the number of eigenvalues. For those infinite layers, we suggest assigning the original learning rates, and excluding them from the layer-wise scheduling. We provided the implementations in commit [cb28648](https://github.com/YefanZhou/TempBalance/commit/cb2864885375e5084d93f48b48f9b9748043daab).

For the performance issue, the results reported in the paper should be able to be reproduced in [experiments](https://github.com/YefanZhou/TempBalance/tree/main#compare-tempbalance-tb-to-cal-and-snr). If you are working on some new scenarios, here are some suggestions: 
- if you are using models with LayerNorm rather than BatchNorm, set the `layernorm` as True, and `batchnorm` as False, when initializing the [class](https://github.com/YefanZhou/TempBalance/blob/f03315317eeebf84b968ffa10675292712bd104d/tempbalance.py#L11). This is added in commit [f033153](https://github.com/YefanZhou/TempBalance/commit/f03315317eeebf84b968ffa10675292712bd104d). This is used for having normalization layer being assigned the same scheduled learning rate as conv/linear layers, and the recent commit includes the two types of normalizations.
- Consider tuning hyperparameters in [class](https://github.com/YefanZhou/TempBalance/blob/f03315317eeebf84b968ffa10675292712bd104d/tempbalance.py#L11): the (`lr_min_ratio`, `lr_max_ratio`) among {(0.6, 1.4), (0.8, 1.2), (1.0, 2.0)}, the `xmin_pos` among {2.0, 1.25}, or tuning original/global learning rate `--lr` [here](https://github.com/YefanZhou/TempBalance_dev/blob/327cf9cee9bb33462b84e6fd4a1c63b29eaee1bb/main_tb.py#L27). 
- If you are working on language datasets or transformer, please refer more to [language_modeling](https://github.com/YefanZhou/TempBalance/tree/main/language_modeling) folder, if you are working on object detection, please refer more to [another folder](https://github.com/YefanZhou/TempBalance/tree/main/object_detection).
